pkg/tests/end_to_end_tests/prometheus_wal_test.go

```diff
@@ -0,0 +1,200 @@
+package end_to_end_tests
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"runtime"
+
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+func generatePrometheusWAL() (string, error) {
+	tmpDir := ""
+
+	if runtime.GOOS == "darwin" {
+		// Docker on Mac lacks access to default os tmp dir - "/var/folders/random_number"
+		// so switch to cross-user tmp dir
+		tmpDir = "/tmp"
+	}
+	dbPath, err := ioutil.TempDir(tmpDir, "prom_dbtest_storage")
+	if err != nil {
+		return "", err
+	}
+	snapPath, err := ioutil.TempDir(tmpDir, "prom_snaptest_storage")
+	if err != nil {
+		return "", err
+	}
+
+	st, err := tsdb.Open(dbPath, nil, nil, &tsdb.Options{
+		RetentionDuration: 15 * 24 * 60 * 60 * 1000, // 15 days in milliseconds
+		NoLockfile:        true,
+	}, nil)
+	if err != nil {
+		return "", err
+	}
+
+	app := st.Appender(context.Background())
+
+	tts := generateRecentLargeTimeseries()
+	if *extendedTest {
+		tts = append(tts, generateRealTimeseries()...)
+	}
+	tts = insertExemplars(tts, 2)
+
+	var ref *uint64
+
+	for _, ts := range tts {
+		ref = nil
+		builder := labels.Builder{}
+
+		for _, l := range ts.Labels {
+			builder.Set(l.Name, l.Value)
+		}
+
+		var (
+			lbls    = builder.Labels()
+			tempRef uint64
+			err     error
+		)
+
+		for _, s := range ts.Samples {
+			if ref == nil || *ref == 0 {
+				tempRef, err = app.Append(tempRef, lbls, s.Timestamp, s.Value)
+				if err != nil {
+					return "", err
+				}
+				ref = &tempRef
+				continue
+			}
+
+			_, err = app.Append(*ref, lbls, s.Timestamp, s.Value)
+
+			if err != nil {
+				return "", err
+			}
+		}
+	}
+
+	if err := app.Commit(); err != nil {
+		return "", err
+	}
+	if err := st.Snapshot(snapPath, true); err != nil {
+		return "", err
+	}
+	if err := os.Mkdir(snapPath+"/wal", 0700); err != nil {
+		return "", err
+	}
+	if err := st.Close(); err != nil {
+		return "", err
+	}
+
+	return dbPath, nil
+}
+
+// generatePrometheusWALWithExemplar generates WAL with exemplars and returns the
+// generated timeseries that can be used to ingest into Promscale, making sure that
```

Comment:
I don't get where you make sure it matches

Reply:
Please see `timeseriesWithExemplars` var in `main_test.go`, and how its used in `exemplar_query_endpoint_test.go`.